.github/workflows/paralleltest.yml

```diff
@@ -1,77 +1,174 @@
-    steps: 
-      - name: Download parallelList - extended.openjdk
-        uses: actions/download-artifact@v2
-        with:
-          name: parallelList.mk
-          path: ./
-
-      - name: Run AQA Parallel Test - extended.openjdk_${{ matrix.test_list }}
-        uses: adoptium/run-aqa@v2
-        with:
-          jdksource: 'customized'
-          customizedSdkUrl: ${{ inputs.customizedSdkUrl }}
-          aqa-testsRepo: 'adoptium/aqa-tests:v0.9.6-release' 
-          build_list: openjdk
-          target : -f parallelList.mk ${{ matrix.test_list }}
-
-      - uses: actions/upload-artifact@v2
-        if: always() # Always run this step (even if the tests failed)
-        with:
-          name: test_output_extended.openjdk_${{ matrix.test_list }}
-          path: ./**/output_*/*.tap
+name: Run AQAvit Paralleltest
+
+on:
+  workflow_dispatch: # Allows the job to be manually triggered
+    inputs:
+      customizedSdkUrl:
+        description: SDK URL
+        required: true
+      aqatest-repo:
+        description: adoptium/aqa-tests repository
+        required: true
+        type: string
+        default: 'adoptium/aqa-tests:v0.9.6-release'
+      testImage:
+        description: Native Libs URL
+        required: true
+      system:
+        description: System being tested
+        required: true
+      pool:
+        description: runner
+        required: true
+        type: string
+        default: None
+
+env:  # Links to the JDK build under test and the native test libs
+  USE_TESTENV_PROPERTIES: true
+
+jobs:
+  configuration:
+    runs-on: ubuntu-latest
+    outputs:
+      agent-config: ${{ steps.selection.outputs.config }}
+    steps:
+      - name: Select agent
+        id: selection
+        run: |
+          if [[ ${{ inputs.pool }} == "macos-x64" ]]; then
+            echo 'config="macos-11"' >> $GITHUB_OUTPUT
+          fi
+          if [[ ${{ inputs.pool }} == "macos-arm64" ]]; then
+            echo "config=['self-hosted', 'macOS', 'ARM64']" >> $GITHUB_OUTPUT
+          fi
+          if [[ ${{ inputs.pool }} == "*windows2019-x64*" ]] || [[ ${{ inputs.pool }} == "*windows2022-aarch64*" ]] || [[ ${{ inputs.pool }} == "*ubuntu20.04-x64*" ]] || [[ ${{ inputs.pool }} == "*ubuntu20.04-aarch64*" ]]; then
+            echo "config=['self-hosted', '1ES.Pool=${{ inputs.pool }}']" >> $GITHUB_OUTPUT
+          fi
+
+  setup-parallel:
+    runs-on: ${{ fromJSON(needs.configuration.outputs.agent-config) }}
+    name: Create AQA parallelList - ${{ inputs.system }}_${{ matrix.suite }}
+    needs: configuration
+    strategy:
+      fail-fast: true
+      matrix:
+        suite: [functional, openjdk, system, perf]
+    steps:
+      - name: Setup windows environment
+        working-directory: ${{ github.workspace }}
+        if: ${{ inputs.system == 'windows' }}
+        shell: pwsh
+        run: |
+          curl -o windows-binaries.zip "${{ secrets.BASE_CONTAINER_URL }}${{ inputs.customizedSdkUrl }}${{ secrets.SAS_TOKEN }}"

```

Comment:
Do we want to use curl or curl.exe? curl will use Invoke-WebRequest while curl.exe is the normal curl.

Reply:
Updated.